code/modules/food_and_drinks/food/snacks_pie.dm

```diff
@@ -41,7 +41,8 @@
 			H.Weaken(1) //splat!
 			H.adjust_blurriness(1)
 			visible_message("<span class='userdanger'>[H] was creamed by [src]!!</span>")
-			H.add_overlay(creamoverlay)
+			if(!(creamoverlay in H.overlays))
+				H.add_overlay(creamoverlay)
```

Comment:
This is not the way to do this, absolutely nothing aside from SSoverlays should be using the overlays var. Make a different var instead